pkg/utils/panics/panics.go

```diff
@@ -0,0 +1,91 @@
+// SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Open Component Model contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package panics
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/open-component-model/ocm/pkg/errors"
+	"github.com/open-component-model/ocm/pkg/logging"
+)
+
+var ReallyCrash = true
+
+type PanicHandler func(interface{}) error
+
+var PanicHandlers = []PanicHandler{logHandler}
+
+// PropagatePanicAsError is intended to be called via defer to
+// map a recovered panic to an error and propagate it to the
+// error return value of the calling function.
+// Use: defer panics.PropagatePanicAsError(&err, false)
+// It incorporates the originally returned error by the surrounding function
+// and all errors provided by the panic handlers plus the mapped recovered panic.
+func PropagatePanicAsError(errp *error, callStandardHandlers bool, additionalHandlers ...PanicHandler) {
+	if r := recover(); r != nil {
+		list := errors.ErrList().Add(mapRecovered(r, 2))
+
+		if callStandardHandlers {
+			// gather errors from standard handler
+			for _, fn := range PanicHandlers {
+				list.Add(fn(r))
+			}
+		}
+
+		// add errors from explicit handlers
+		for _, fn := range additionalHandlers {
+			list.Add(fn(r))
+		}
+
+		*errp = list.Result()
+	}
+}
+
+func HandlePanic(additionalHandlers ...PanicHandler) {
+	if r := recover(); r != nil {
+		for _, fn := range PanicHandlers {
+			_ = fn(r)
+		}
+
+		for _, fn := range additionalHandlers {
+			_ = fn(r)
+		}
+
+		if ReallyCrash {
+			panic(r)
+		}
+	}
+}
+
+// RegisterPanicHandler adds handlers to the panic handler.
+func RegisterPanicHandler(handler PanicHandler) {
+	PanicHandlers = append(PanicHandlers, handler)
+}
+
+func mapRecovered(r interface{}, offset int) error {
```

Comment:
What is the purpose of `offset` and the append lines? 🤔 That is way too much magic. :) I'm sure there is a simpler way of handling that Join.